components/ActionSheet/src/MDCActionSheetController.h

```diff
@@ -0,0 +1,182 @@
+
+ After creating the alert controller, add actions to the controller by calling -addAction.
+
+ @param title The title of the alert.
+ @param message Descriptive text that summarizes a decision in a sentence or two.
+ @return An initialized MDCActionSheetController object.
+ */
++ (nonnull instancetype)actionSheetControllerWithTitle:(nullable NSString *)title
+                                               message:(nullable NSString *)message;
+
+/**
+ Convenience initializer to create and return a view controller for displaying an alert to the user.
+
+ After creating the alert controller, add actions to the controller by calling -addAction.
+
+ @param title The title of the alert.
+ @return An initialized MDCActionSheetController object.
+ */
++ (instancetype)actionSheetControllerWithTitle:(nullable NSString *)title;
+
+/**
+ Action sheet controllers must be created with actionSheetControllerWithTitle: or
+ with actionSheetControllerWithTitle:message:
+ */
+- (nonnull instancetype)initWithNibName:(NSString *)nibNameOrNil
+                                 bundle:(NSBundle *)nibBundleOrNil NS_UNAVAILABLE;
+
+/**
+ Action sheet controllers must be created with actionSheetControllerwithTitle:
+   or with actionSheetControllerWithTitle:message:
+ */
+- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder NS_UNAVAILABLE;
+
+/**
+ Adds an action to the action sheet.
+
+ Actions are the possible reactions of the user to the presented alert. Actions are added as a
+ list item in the list of the action sheet.
+ Action buttons will be laid out from top to bottom depending on the order they
+ were added.
+
+ @param action Will be added to the end of MDCActionSheetController.actions.
+ */
+- (void)addAction:(nonnull MDCActionSheetAction *)action;
+
+/**
+ The actions that the user can take in response to the action sheet.
+
+ The order of the actions in the array matches the order in which they were added to the action sheet.
+ */
+@property (nonatomic, nonnull, readonly) NSArray<MDCActionSheetAction *> *actions;
+
+/**
+ The title of the action sheet controller.
+
+ If this is updated after presentation the view will be updated to match the new value.
+ */
+@property (nonatomic, nullable, copy) NSString *title;
+
+/**
+ The message of the action sheet controller.
+
+ If this is updated after presentation the view will be updated to match the new value.
+ */
+@property (nonatomic, nullable, copy) NSString *message;
+
+/*
+ Indicates whether the button should automatically update its font when the device’s
+ UIContentSizeCategory is changed.
+
+ This property is modeled after the adjustsFontForContentSizeCategory property in the
+ UIConnectSizeCategoryAdjusting protocol added by Apple in iOS 10.0.
+
+ If set to YES, this button will base its text font on MDCFontTextStyleButton.
+
+ Defaults value is NO.
+ */
+@property(nonatomic, readwrite, setter=mdc_setAdjustsFontForContentSizeCategory:)
+    BOOL mdc_adjustsFontForContentSizeCategory;
+
```

Comment:
Should there be APIs for theming? From the [Material Guidelines theming section](https://material.io/design/components/sheets-bottom.html#theming) it looks like color and shape (not yet built on iOS) theming are permitted.

Suggestion:
```objective
@property(nonatomic, strong) UIColor *titleTextColor;
@property(nonatomic, strong) UIFont *titleTextFont;
@property(nonatomic, strong) UIColor *messageTextColor;
@property(nonatomic, strong) UIFont *messageTextFont;
@property(nonatomic, strong) UIColor *actionTextColor;
@property(nonatomic, strong) UIColor *actionTintColor; // for template image tinting
@property(nonatomic, strong) UIColor *inkColor; // Overlay ripple color for touch feedback
```

Reply:
I am going to add all of those API's at a later date.